File: app/src/main/java/com/laundromat/pos/data/model/PosUser.kt
package com.laundromat.pos.data.model

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "pos_users",
    indices = [Index(value = ["username"], unique = true)]
)
data class PosUser(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val username: String,
    val displayName: String,
    val pin: String,
    val role: String,
    val isActive: Boolean = true,
    val lastLoginAt: Long? = null
)

File: flutter_app/lib/src/models/pos_user.dart
class PosUser {
  const PosUser({
    required this.id,
    required this.username,
    required this.displayName,
    required this.pin,
    required this.role,
  });

  final int id;
  final String username;
  final String displayName;
  final String pin;
  final String role;
}
